cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/kv/HBaseKVTableDefinition.java

```diff
@@ -70,23 +71,23 @@ public DatasetAdmin getAdmin(DatasetSpecification spec, ClassLoader classLoader)
   @Override
   public NoTxKeyValueTable getDataset(DatasetSpecification spec,
                                       Map<String, String> arguments, ClassLoader classLoader) throws IOException {
-    return new KVTableImpl(spec.getName(), hConf);
+    return new KVTableImpl(spec.getName(), hConf, tableUtil);
```

Comment:
so the name in the spec is the namespaced name and not the user provided name? It's very confusing where in the code a name is namespaced vs. not, though I think you mentioned there is a bigger jira for that.